src/dawn_native/vulkan/CommandBufferVk.cpp

```diff
@@ -53,16 +53,19 @@ namespace dawn_native { namespace vulkan {
 
             region.imageSubresource.aspectMask = texture->GetVkAspectMask();
             region.imageSubresource.mipLevel = textureLocation.level;
-            region.imageSubresource.baseArrayLayer = 0;
+
+            // TODO(jiawei.shao@intel.com): support 1D and 3D textures
+            ASSERT(texture->GetDimension() == dawn::TextureDimension::e2D);
```

Comment:
drive-by: I'd prefer if this section looked a little more like we might expect in final code, e.g.:
```cpp
if (texture->GetDimension() == dawn::TextureDimension::e2D) {
  // set the appropriate fields of `region`
} else {
  ASSERT(false); // or something; I don't think we have a NOTIMPLEMENTED but you could add one
}```

Reply:
(Same comment for other similar places in this pull request)